pkg/products/threescale/reconciler.go

```diff
@@ -418,40 +418,72 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8scl
 			Name:      apiManagerName,
 			Namespace: r.Config.GetNamespace(),
 		},
-	}
-
-	status, err := controllerutil.CreateOrUpdate(ctx, serverClient, apim, func() error {
-
-		apim.Spec = threescalev1.APIManagerSpec{
-			HighAvailability: &threescalev1.HighAvailabilitySpec{
-				Enabled: true,
-			},
+		Spec: threescalev1.APIManagerSpec{
```

Comment:
Why do we populate all the spec here, and then do it again in the createOrUpdate method? Also, are the values read into the object in the mutator function, or will they always come through as zero because of this block?

Reply:
If we don't have this here the createOrUpdate fails because the objects are nil on initial install, When it gets to the createOrUpdate the values are updated so we can compare required vs current values